client/languages.ts

```diff
@@ -0,0 +1,23 @@
+// languages.ts
+
+// giving a name might help comprehension
+// I'm pretty sure that the `languages.js` file
+// is autogenerated otherwise we could include
+// the actual slugs for extra type safety - here
+// it's really not needed or helpful though
+type LanguageSlug = string;
+type WPLocale = string;
+
+type BaseLanguage = {
+	langSlug: LanguageSlug;
+	name: string;
+	popular?: number;
+	rtl?: boolean;
+	territories: string[];
+	value: number;
+	wpLocale: WPLocale | '';
+};
+
+type SubLanguage = BaseLanguage & { parentLangSlug: string };
+
+export type Language = BaseLanguage | SubLanguage;
```

Comment:
ah…I was speaking specifically about using the existing `languages.js` and renaming it to `languages.ts` but maybe this works too?

Reply:
maybe it's best not to mingle the type declaration and the huge list of languages. in that case I don't know. if this works then maybe that's best